lib/modules/dosomething/dosomething_signup/includes/Signup.php

```diff
@@ -89,29 +89,8 @@ private function build($data) {
     $this->id = $data->sid;
     $this->created_at = $data->timestamp;
 
-    $drupal_user = user_load($data->uid);
-
-    if (!isset($data->field_northstar_id_value) || $data->field_northstar_id_value === 'NONE') {
-      $northstar_response = dosomething_northstar_get_northstar_user($data->uid);
-      $northstar_response = json_decode($northstar_response);
-
-      if (!empty($northstar_response->data) && !isset($northstar_response->error)) {
-        $northstar_user = $northstar_response->data;
-        user_save($drupal_user, ['field_northstar_id' => [LANGUAGE_NONE => [0 => ['value' => $northstar_user->id]]]]);
-      }
-    }
-
-    $last_initial = substr(dosomething_helpers_extract_field_data($drupal_user->field_last_name), 0, 1);
-    if ($last_initial === FALSE) {
-      $last_initial = NULL;
-    }
-
     $this->user = [
```

Comment:
Yep, so Phoenix would always return _just_ `drupal_id` for the user response since that's all it can reliably say about a user. And then for any requests going through the Northstar proxy, Northstar would add the relevant `first_name`, `last_initial`, etc etc to each `user` object.